Update Cloud docs for local runs and plan updates (#6642)
dagger/dagger: docs/current_docs/cloud/572923-get-started.mdx

This guide helps you get started with Dagger Cloud. Here are the steps you'll follow:

- [Sign up for Dagger Cloud](#step-1-sign-up-for-dagger-cloud)
- [Send telemetry to Dagger Cloud](#step-2-send-telemetry-to-dagger-cloud)
- [Visualize pipeline runs with Dagger Cloud](#step-3-visualize-your-pipeline-runs-with-dagger-cloud)

> - [Synchronize cache volumes with the Dagger Cloud experimental distributed cache (available on the Team plan)](#step-4-use-cache-volumes-with-the-experimental-dagger-cloud-cache)

Reviewer: This list item doesn't match the heading title for the step.

> - [Synchronize cache volumes with the Dagger Cloud experimental distributed cache (available on the Team plan)](#step-4-use-cache-volumes-with-the-experimental-dagger-cloud-cache)

Reviewer: ```suggestion
- [Use cache volumes with the experimental Dagger Cloud cache](#step-4-use-cache-volumes-with-the-experimental-dagger-cloud-cache)
```

> - [Synchronize cache volumes with the Dagger Cloud experimental distributed cache (available on the Team plan)](#step-4-use-cache-volumes-with-the-experimental-dagger-cloud-cache-team-plan)

Reviewer 2: ```suggestion
- [Synchronize cache volumes with the Dagger Cloud experimental distributed cache (available on the Team plan)](#step-4-use-cache-volumes-with-the-experimental-dagger-cloud-cache)
```

Follow the steps below to sign up for Dagger Cloud, create an organization and obtain a Dagger Cloud token.

1. Sign up for Dagger Cloud by selecting a plan on the [Dagger website](https://www.dagger.io/pricing). Click *Continue with GitHub* to log in with your GitHub account.

1. After authorizing Dagger Cloud for your GitHub account, you'll create your organization.

    :::info Naming your organization
    Organization names may contain alphanumeric characters and dashes and are unique across Dagger Cloud. We recommend using your company name or team name for your organization.
    :::

> 1. Review the available Dagger Cloud subscription plans. Choose a plan by clicking *Select*.

Author: Thanks! I'll change it to: "Review the Dagger Cloud subscription plans and select one to move forward." Or an equivalent.

> 1. If you selected the *Team* plan:

Reviewer 2: I don't think `*Team*` actually makes it bold since in the page you can barely notice the difference :thinking: 
![image](https://github.com/dagger/dagger/assets/1578458/475c8374-f574-4777-ab95-c891e96b892c)

Author: Yeah, that's making them italicized. I'm updating them to **text** in a commit coming in soon.

    1. You will be presented with the option to add teammates to your Dagger Cloud account. This step is optional and not available in the Individual plan.
    
    1. You will then enter your payment information. After your free 14 day trial completes, you will be automatically subscribed to the Dagger Cloud Team plan.

> 1. Click *Go to dashboard*. Your dashboard will initially be blank. The next step walks you through how to send telemetry to Dagger Cloud.

Reviewer 2: nit: Your dashboard will initially be blank. Do we need to be redundant on that? The user will see a similar message after landing to that page. Might not be necessary to re-iterate that here as well.

Author: Fixed.

## Step 2: Send telemetry to Dagger Cloud

:::info
At the end of this step, you will have connected Dagger Cloud with your CI provider or CI tool using your Dagger Cloud token. If you have already connected Dagger Cloud with your CI provider/tool and don't see data yet in your dashboard, skip to the next step.
:::

You can use Dagger Cloud with your CI and for your local development workflows. You instruct your Dagger Engine to connect to Dagger Cloud by referencing your unique Dagger Cloud token. Dagger Cloud creates this token automatically when you sign up.

To find your token, browse to the **Organization Settings** page, by clicking your profile photo in the left sidebar and then clicking Organization Settings. Select the **Configuration** tab. You can also use this URL pattern: `https://dagger.cloud/{Your Org Name}/settings?tab=configuration`

    ![Get token](/img/current_docs/cloud/get-started/get-token.png)

  :::warning

>   If you regenerate your token, you must replace the token wherever you've referenced it. To reduce operational interruptions, only regenerate your token if it has been leaked. 

Reviewer: ```suggestion
  If you regenerate your token, you must replace the token wherever you've referenced it. To reduce operational interruptions, only regenerate your token if it has leaked. 
```

  :::

Once you have your token, you can now use it to send telemetry from your CI and development machines.

### Local development
You can visualize and debug your local Dagger pipeline runs with Dagger Cloud to identify issues before pushing them to CI.

To configure local development, set your Dagger Cloud token as an environment variable. For example: `export DAGGER_CLOUD_TOKEN={your token}`

:::info BEST PRACTICES FOR VISUALIZING LOCAL RUNS IN DAGGER CLOUD

> Dagger Cloud uses metadata from your version control system (VCS), such as git, to populate some information in the user interface. Therefore, you should run your Dagger pipelines within a VCS context, such as a local directory linked to a cloned git repository.

Reviewer: ```suggestion
Dagger Cloud uses metadata from your version control system (VCS), such as Git, to populate some information in the user interface. Therefore, you should run your Dagger pipelines within a VCS context, such as a local directory linked to a cloned Git repository.
```

:::

### In CI
The general procedure to connect Dagger Cloud with a CI provider/CI tool is:

- Add your Dagger Cloud token to your CI workflows.
  - Store the Dagger Cloud token as a secret with your CI.
    :::danger Keep your Dagger Cloud token private
    You must store the Dagger Cloud token as a secret (not plaintext) with your CI provider and reference it in your CI’s workflow. Using a secret is recommended to protect your Dagger Cloud account from being used by forks of your project. We provide links in the steps below for configuring secrets with popular CI tools.
    :::
  - Add the secret to your CI environment as a variable named `DAGGER_CLOUD_TOKEN`.

> - For **Team** plan subscribers with ephemeral CI runners only: Add a wait step (example below) for the Docker configuration to allow the Dagger Engine to push cache volumes to the experimental Dagger Cloud distributed cache. Subscribers on the **Individual** plan do not have access to the experimental distributed cache and do not need to add this step.

Reviewer: Note that this step is missing for Jenkins and Argo tabs. Do we need to add it, or is its absence intentional?

> - For *Team* plan subscribers with ephemeral CI runners only: Add a wait step for the Docker configuration to allow the Dagger Engine to push cache volumes to the experimental Dagger Cloud distributed cache. Subscribers on the *Individual* plan do not have access to the experimental distributed cache and do not need to add this step.

Reviewer 2: ```suggestion
- For *Team* plan subscribers with ephemeral CI runners only: Add a wait step (example below) for the Docker configuration to allow the Dagger Engine to push cache volumes to the experimental Dagger Cloud distributed cache. Subscribers on the *Individual* plan do not have access to the experimental distributed cache and do not need to add this step.
```

Author: Updated in the 4 instances where this is mentioned

- If you are using GitHub Actions, install the Dagger Cloud GitHub app for GitHub Checks. This app adds a GitHub Check to your pull requests and links to the run status and pipeline visualization for the commit in Dagger Cloud.

You can use Dagger Cloud whether you're hosting your own CI runners and infrastructure or using hosted/SaaS runners.

> :::danger

Reviewer 2: Unrelated to this PR, this danger note is a bit confusing. Shall we revisit this one @vikram-dagger ?

Reviewer: AFAIR we added it because it was a point of concern for AWS users.

</TabItem>
</Tabs>

## Step 3: Visualize your pipeline runs with Dagger Cloud

:::info
At the end of this step, you will have data from one or more CI or local runs available for inspection and debugging in Dagger Cloud.
:::

### Local development
:::info Best practices for visualizing local runs in Dagger Cloud
Dagger Cloud uses metadata from your version control system (VCS), such as git, to populate some information in the user interface. Therefore, you should run your Dagger pipelines within a VCS context, such as a local directory linked to a cloned git repository.
:::

Before completing these steps, ensure you've [added your Dagger Cloud token as an environment variable](#local-development). 

1. In your terminal, run your Dagger pipeline. If your environment variable is properly defined, you will see a Dagger Cloud URL printed in the terminal. 
1. Click on the URL to view your pipeline run in Dagger Cloud.

> Here's an example running a Dagger NodeJS pipeline named index.mjs locally after committing a change with git.

Reviewer: ```suggestion
Here's an example running a Dagger Node.js pipeline named `index.mjs` locally after committing a change with Git.
```

```shell
  git commit -s -m "Updated quickstart link in README"
  dagger run node index.mjs
  ```

> Because we ran this in a git context, the Dagger Cloud UI shows the VCS git metadata associated with the commit. 

Reviewer: ```suggestion
Because we ran this in a Git context, the Dagger Cloud UI shows the VCS Git metadata associated with the commit. 
```

> (ADD SCREENSHOT OF LOCAL RUN)

Reviewer 2: missing screenshot. Just highlighting so we don't miss this.

If you run outside of a git context, Dagger Cloud will show the same detailed steps. However, the top-level metadata will not be populated. 

> (ADD SCREENSHOT OF LOCAL RUN)

Reviewer 2: missing screenshot. Just highlighting so we don't miss this.

> If you run outside of a git context, Dagger Cloud will show the same detailed steps. However, the top-level metadata will not be populated. 

Reviewer: ```suggestion
If you run outside of a Git context, Dagger Cloud will show the same detailed steps. However, the top-level metadata will not be populated. 
```

> :::info Multiple Dagger Cloud organizations

Reviewer: This is an odd place for this note. Perhaps move it to the end of the section about generating a token, or in the reference page?

Author: We're merging this ASAP. Let's pull any feedback into a new issue.

Learn more about the [Dagger Cloud user interface](./reference/741031-user-interface.mdx).

## Step 4: Use cache volumes with the experimental Dagger Cloud cache

:::info

> At the end of this step, you will have integrated Dagger Cloud's experimental distributed cache with your workflows.

Reviewer 2: nit: highlight experimental?

:::

:::info
Dagger Cloud's distributed caching feature is only available under the *Team* plan.
:::

One of Dagger's most powerful features is its ability to [cache data across pipeline runs](../quickstart/635927-caching.mdx). This is especially useful when dealing with package managers such as `npm`, `maven`, `pip` and similar. For these tools to cache properly, they need their own cache data (usually a directory) to be persisted between runs. Since these dependencies are usually locked to specific versions in the application's manifest, re-downloading them on every pipeline run is inefficient and time-consuming.

Dagger comes with built-in support to define one or more such directories as cache volumes and persist their contents across runs. Dagger Cloud enhances caching support significantly and allows cross-host synchronization of cache volumes. This allows multiple machines, including ephemeral runners, to intelligently share a distributed cache.

> Dagger Cloud automatically detects and creates cache volumes when they are declared in your code. To see how this works, add a cache volume to your Dagger pipeline and then trigger a run. If you're using the starter application and Dagger pipeline from [Appendix A](#appendix-a-create-a-dagger-pipeline), do this by updating the Dagger pipeline code as shown below (changes are highlighted):

Reviewer 2: I think we should revisit this paragraph in general as it puts too much emphasis on cache volumes vs caching operations. Let's put that in the queue @vikram-dagger

Reviewer: I didn't understand this @marcosnils . What change are you suggesting? That we should not mention cache volumes here and only focus on the experimental cache?
